dace/codegen/dispatcher.py

```diff
@@ -120,7 +137,7 @@ def __init__(self, framecode):
         # type: Dict[dace.dtypes.InstrumentationType, InstrumentationProvider]
         self.instrumentation = {}
 
-        self._array_dispatchers = {
+        self._array_dispatchers : dict[dtypes.StorageType, target.TargetCodeGenerator] = {
```

Comment:
I know that this is inconsistent with the project but all generic concrete collections are now deprecated in favor of the builtins (see, for example, https://docs.python.org/3/library/typing.html#typing.Dict).

Reply:
since Python 3.9 yes, but you can't use it with older versions of python. as long as we support 3.6+, we have to use Dict. I can't wait for this either.